src/Trace/Tracer/ExtensionTracer.php

```diff
@@ -0,0 +1,165 @@
+ */
+
+namespace Google\Cloud\Trace\Tracer;
+
+use Google\Cloud\Core\ArrayTrait;
+use Google\Cloud\Trace\TraceContext;
+use Google\Cloud\Trace\TraceSpan;
+
+/**
+ * This implementation of the TracerInterface utilizes the stackdriver extension
+ * to manage span context. The stackdriver extension augments user created spans and
+ * adds automatic tracing to several commonly desired events.
+ */
+class ExtensionTracer implements TracerInterface
+{
+    use ArrayTrait;
+
+    /**
+     * Create a new ExtensionTracer
+     *
+     * @param TraceContext $context [optional] The TraceContext to begin with. If none
+     *      provided, a fresh TraceContext will be generated.
+     */
+    public function __construct(TraceContext $context = null)
+    {
+        $context = $context ?: new TraceContext();
+        stackdriver_trace_set_context($context->traceId(), (int) $context->spanId());
+    }
+
+    /**
+     * Instrument a callable by creating a TraceSpan
+     *
+     * @param array $spanOptions Options for the span.
+     *      {@see Google\Cloud\Trace\TraceSpan::__construct()}
+     * @param callable $callable The callable to inSpan.
+     * @param array $arguments [optional] Arguments for the callable.
+     * @return mixed The result of the callable
+     */
+    public function inSpan(array $spanOptions, callable $callable, array $arguments = [])
+    {
+        $this->startSpan($spanOptions);
+        try {
+            return call_user_func_array($callable, $arguments);
+        } finally {
+            $this->endSpan();
+        }
+    }
+
+    /**
+     * Start a new Span. The start time is already set to the current time.
+     *
+     * @param array $spanOptions [optional] Options for the span.
+     *      {@see Google\Cloud\Trace\TraceSpan::__construct()}
+     */
+    public function startSpan(array $spanOptions)
+    {
+        $name = $this->pluck('name', $spanOptions, false) ?: $this->generateSpanName();
+        stackdriver_trace_begin($name, $spanOptions);
+    }
+
+    /**
+     * Finish the current context's Span.
+     *
+     * @return bool
+     */
+    public function endSpan()
+    {
+        return stackdriver_trace_finish();
+    }
+
+    /**
+     * Return the current context.
+     *
+     * @return TraceContext
+     */
+    public function context()
+    {
+        $context = stackdriver_trace_context() + [
+            'spanId' => null
+        ];
+        return new TraceContext(
+            $context['traceId'],
+            $context['spanId']
+        );
+    }
+
+    /**
+     * Return the spans collected.
+     *
+     * @return TraceSpan[]
+     */
+    public function spans()
+    {
+        return array_map(function ($span) {
+            return new TraceSpan($span);
+        }, stackdriver_trace_list());
+    }
+
+    /**
+     * Add a label to the current TraceSpan
+     *
+     * @param string $label
+     * @param string $value
+     */
+    public function addLabel($label, $value)
+    {
```

Comment:
There will be, it's not yet implemented in the version being developed. I can remove this tracer for now.